lib/metadata/include/irods/metadata.hpp

```diff
@@ -0,0 +1,79 @@
+#ifndef IRODS_METADATA_HPP
+#define IRODS_METADATA_HPP
+
+#include <string>
+#include <vector>
+#include <map>
+
+#include "irods/entity.hpp"
+struct RsComm;
+struct RcComm;
```

Comment:
Move these forward declarations into the appropriate `#ifdef`-block (notice the `using` directives).